### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by jamesarosen**
1. move `sprockets/engine` to `sprockets/commonjs/engine` so it doesn't conflict with any other Sprockets Engines
2. add `Sprockets::CommonJS.default_namespace=` so clients don't have to monkey-patch to change it
3. make `Sprockets::CommonJS#prepare` and `#evaluate` protected, like they are for `Tilt::Template` (the parent class)
4. make helper method `Sprockets::CommonJS#namespace` private
5. extract helper method `Sprockets::CommonJS#common_js_module?`
6. extract the CommonJS module definition JavaScript into a constant

### CODE REVIEW

File: lib/sprockets/commonjs.rb [hunk 1]
```diff
@@ -3,35 +3,44 @@
 
 module Sprockets
   class CommonJS < Tilt::Template
-    self.default_mime_type = 'application/javascript'
 
-    def self.default_namespace
-      'this.require'
+    DEFINE_WRAPPER = '%s.define({%s:' +
```
Comment (maccman) [inline lib/sprockets/commonjs.rb:7]: Maybe use `%w{}`

File: lib/sprockets/commonjs.rb [hunk 2]
```diff
@@ -3,35 +3,44 @@
...
         data
       end
     end
+
+    private
+
+    attr_reader :namespace
+
+    def common_js_module?(scope)
+      scope.pathname.basename.to_s.include?('.module')
```
Comment (maccman) [inline lib/sprockets/commonjs.rb:40]: Maybe call `commonjs_module?`

**Comment by maccman** [comment]
Awesome - few nitpicks, but am keen to get this merged.

Also, just want to make sure you're aware, but there's an edge branch for the latest version of Sprockets: https://github.com/maccman/sprockets-commonjs/tree/edge

File: lib/sprockets/commonjs.rb (hunk 1 above)
Comment (jamesarosen) [inline lib/sprockets/commonjs.rb:7]: If we use `%w{}` or `<<-`, we'll get extra whitespace. Is that OK?

Comment (maccman) [inline lib/sprockets/commonjs.rb:7]: Good point!

**Comment by jamesarosen** [comment]
I'm happy to do this against the edge branch as well.

**Comment by maccman** [comment]
That would be awesome.

**Pull request merged**